### PR TITLE
feat: add filter to getAll Pizza

### LINKED ISSUE #47: Sprint 03 - Create Pizza getAll with filters

### PR DESCRIPTION
**Pull request by lucaspere**
This PR adds a filter to ``getAll`` in ``Pizza`` model.

#### Changelog

- Creates ``buildLikeFilter``, ``buildValorFilter`` and ``buildWhere`` pizza helpers;
- Refactor the ``getAll``.

### CODE REVIEW

**Comment by Lucas-Angelo** [comment]
@lucaspere can you add postman collection with request and filters for Pizza? Or, cURL. For me to test

**Review by Lucas-Angelo — APPROVED** [review]

File: src/backend/services/PizzaService.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 const AppError = require("../errors/AppError");
 const Pizza = require("../models/Pizza");
-const { SortPaginate } = require("../helpers/SortPaginate");
+const { SortPaginate, buildWhere } = require("../helpers/SortPaginate");
```
Comment (guizombas) [inline src/backend/services/PizzaService.js]: buildWhere is being imported from the wrong file

File: src/backend/helpers/pizza.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        created_at_start,
+        created_at_end,
+        updated_at_start,
+        updated_at_end,
+        ...others
+    } = query;
+
+    const where = Object.entries(others).reduce((query, [key, value]) => {
+        query[key] =
+            typeof value == "number"
```
Comment (guizombas) [inline src/backend/helpers/pizza.js]: will this work? as the value comes from the query string, won't it be of type "string" for every case ?

**Review by guizombas — CHANGES_REQUESTED** [review]

File: src/backend/services/PizzaService.js (hunk 1 above)
Comment (lucaspere) [reply src/backend/services/PizzaService.js]: Good catch!

**Review by guizombas — CHANGES_REQUESTED** [review]
getting an error when trying using the 'valor_maior_que' or 'valor_menor_que' filter, because it's not a valid filed name. 

image

```
http://localhost:3001/api/pizza?valor_maior_que=9

response:

{
  "message": "Erro interno do servidor!",
  "error": {
    "name": "SequelizeDatabaseError",
    "parent": {
      "code": "ER_BAD_FIELD_ERROR",
      "errno": 1054,
      "sqlState": "42S22",
      "sqlMessage": "Unknown column 'Pizza.valor_maior_que' in 'where clause'",
      "sql": "SELECT count(*) AS `count` FROM `pizza` AS `Pizza` WHERE `Pizza`.`valor_maior_que` >= '9' AND `Pizza`.`updated_at` BETWEEN '1970-01-01 00:00:00' AND '2022-10-18 18:39:25' AND `Pizza`.`created_at` BETWEEN '1970-01-01 00:00:00' AND '2022-10-18 18:39:25';"
    },
    "original": {
      "code": "ER_BAD_FIELD_ERROR",
      "errno": 1054,
      "sqlState": "42S22",
      "sqlMessage": "Unknown column 'Pizza.valor_maior_que' in 'where clause'",
      "sql": "SELECT count(*) AS `count` FROM `pizza` AS `Pizza` WHERE `Pizza`.`valor_maior_que` >= '9' AND `Pizza`.`updated_at` BETWEEN '1970-01-01 00:00:00' AND '2022-10-18 18:39:25' AND `Pizza`.`created_at` BETWEEN '1970-01-01 00:00:00' AND '2022-10-18 18:39:25';"
    },
    "sql": "SELECT count(*) AS `count` FROM `pizza` AS `Pizza` WHERE `Pizza`.`valor_maior_que` >= '9' AND `Pizza`.`updated_at` BETWEEN '1970-01-01 00:00:00' AND '2022-10-18 18:39:25' AND `Pizza`.`created_at` BETWEEN '1970-01-01 00:00:00' AND '2022-10-18 18:39:25';",
    "parameters": {
      
    }
  }
}
```

this bug also happens when you pass any other query string parametter, like 'order' and 'atribute', required for sorting the data: 

image

**Review by guizombas — APPROVED** [review]

**Pull request merged**